### ISSUE TITLE
Invoking Settings / Edit when the editor option is something other than default hits an exception

### ISSUE THREAD

**Issue by mortonfox** [fitztrev/shuttle#137]
It happens in Shuttle v1.2.4.
For example, when I have "editor": "vim" in my .shuttle.json, I see this come up in Console:
```
10/22/2015 3:55:09.458 PM Shuttle[4427]: An uncaught exception was raised
10/22/2015 3:55:09.461 PM Shuttle[4427]: *** -[__NSPlaceholderArray initWithObjects:count:]: attempt to insert nil object from objects[1]
10/22/2015 3:55:09.466 PM Shuttle[4427]: (
	0   CoreFoundation                      0x00007fff8fe3d25c __exceptionPreprocess + 172
	1   libobjc.A.dylib                     0x00007fff922f7e75 objc_exception_throw + 43
	2   CoreFoundation                      0x00007fff8fcf59a6 -[__NSPlaceholderArray initWithObjects:count:] + 358
	3   CoreFoundation                      0x00007fff8fd59440 +[NSArray arrayWithObjects:count:] + 48
	4   Shuttle                             0x00000001065f16eb -[AppDelegate openHost:] + 1460
	5   Shuttle                             0x00000001065f22a0 -[AppDelegate configure:] + 209
	6   AppKit                              0x00007fff86e67260 -[NSApplication sendAction:to:from:] + 327
	7   AppKit                              0x00007fff86e821c8 -[NSMenuItem _corePerformAction] + 394
	8   AppKit                              0x00007fff86e81f04 -[NSCarbonMenuImpl performActionWithHighlightingForItemAtIndex:] + 117
	9   AppKit                              0x00007fff86ed140d -[NSMenu _internalPerformActionForItemAtIndex:] + 35
	10  AppKit                              0x00007fff86ed1289 -[NSCarbonMenuImpl _carbonCommandProcessEvent:handlerCallRef:] + 104
	11  AppKit                              0x00007fff86e77ff6 NSSLMMenuEventHandler + 716
	12  HIToolbox                           0x00007fff90c3f1d4 _ZL23DispatchEventToHandlersP14EventTargetRecP14OpaqueEventRefP14HandlerCallRec + 892
	13  HIToolbox                           0x00007fff90c3e787 _ZL30SendEventToEventTargetInternalP14OpaqueEventRefP20OpaqueEventTargetRefP14HandlerCallRec + 385
	14  HIToolbox                           0x00007fff90c52880 SendEventToEventTarget + 40
	15  HIToolbox                           0x00007fff90c88640 _ZL18SendHICommandEventjPK9HICommandjjhPKvP20OpaqueEventTargetRefS5_PP14OpaqueEventRef + 420
	16  HIToolbox                           0x00007fff90cbb228 SendMenuCommandWithContextAndModifiers + 59
	17  HIToolbox                           0x00007fff90cbb1d0 SendMenuItemSelectedEvent + 178
	18  HIToolbox                           0x00007fff90cbb0af _ZL19FinishMenuSelectionP13SelectionDataP10MenuResultS2_ + 94
	19  HIToolbox                           0x00007fff90c9919e _ZL19PopUpMenuSelectCoreP8MenuData5PointdS1_tjPK4RecttjS4_S4_PK10__CFStringPP13OpaqueMenuRefPt + 1726
	20  HIToolbox                           0x00007fff90c982f6 _HandlePopUpMenuSelection7 + 596
	21  AppKit                              0x00007fff86ed0161 _NSSLMPopUpCarbonMenu3 + 4153
	22  AppKit                              0x00007fff86ecf116 _NSPopUpCarbonMenu3 + 39
	23  AppKit                              0x00007fff86ecedd6 -[NSCarbonMenuImpl popUpMenu:atLocation:width:forView:withSelectedItem:withFont:withFlags:withOptions:] + 350
	24  AppKit                              0x00007fff872c0729 +[NSStatusBarButtonCell popupStatusBarMenu:inRect:ofView:withEvent:] + 699
	25  AppKit                              0x00007fff86f8db60 -[NSStatusBarButtonCell trackMouse:inRect:ofView:untilMouseUp:] + 153
	26  AppKit                              0x00007fff86ecc13d -[NSControl mouseDown:] + 706
	27  AppKit                              0x00007fff86e4da58 -[NSWindow sendEvent:] + 11296
	28  AppKit                              0x00007fff86f4911b -[NSStatusBarWindow sendEvent:] + 247
	29  AppKit                              0x00007fff86dec5d4 -[NSApplication sendEvent:] + 2021
	30  AppKit                              0x00007fff86c3c9f9 -[NSApplication run] + 646
	31  AppKit                              0x00007fff86c27783 NSApplicationMain + 940
	32  Shuttle                             0x00000001065ee260 start + 52
	33  ???                                 0x0000000000000001 0x0 + 1
)
```

**Comment by thshdw** [comment]
@mortonfox do you get the exception when you try to click on edit? or do you get the execption when you run a commnad. If on command is it all commands or just one?

**Comment by thshdw** [comment]
@mortonfox Fixed in v1.2.5 Thanks for reporting this issue! :+1:
